internal/awslambda/extension/manager_integration_test.go

```diff
@@ -0,0 +1,177 @@
+package extension
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/ash2k/stager/wait"
+	"github.com/gorilla/mux"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/atlassian/gostatsd/internal/awslambda/extension/api"
+	"github.com/atlassian/gostatsd/internal/awslambda/extension/telemetry"
+	"github.com/atlassian/gostatsd/internal/flush"
+	"github.com/atlassian/gostatsd/pb"
+	"github.com/atlassian/gostatsd/pkg/statsd"
+	"github.com/atlassian/gostatsd/pkg/transport"
+)
+
+func TestManagerGostatsDIntegrationTest(t *testing.T) {
+	t.Parallel()
+
+	var forwarderCallReceived uint64
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	readyCh := make(chan struct{}, 1)
+
+	apiMux := mux.NewRouter()
+	apiMux.Handle("/v2/raw", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		atomic.AddUint64(&forwarderCallReceived, 1)
+		buf, errr := io.ReadAll(r.Body)
+		require.NoError(t, errr, "Must not error when reading request")
+
+		var data pb.RawMessageV2
+		require.NoError(t, proto.Unmarshal(buf, &data))
+
+		if len(data.GetCounters()) == 0 {
+			readyCh <- struct{}{}
+			close(readyCh)
+		}
+		for _, v := range data.GetCounters() {
+			for tag, m := range v.TagMap {
+				assert.EqualValues(t, 42, m.Value)
+				assert.True(t, strings.Contains(tag, "fruit:pineapple"))
+				cancel()
+			}
+		}
+	}))
+	apiMux.Handle("/v2/event", http.HandlerFunc(func(_ http.ResponseWriter, _ *http.Request) {}))
+
+	gsdServer := httptest.NewServer(apiMux)
+	t.Cleanup(gsdServer.Close)
+
+	log := logrus.New()
```

Comment:
This is going to be rather noisey, is there way to provide it a test logger?

Reply:
Setting the output to `io.Discard`